README.md

```diff
@@ -1,27 +1,176 @@
+    ...clientOptions,
+    skipContactPublishing: true,
+    persistConversations: false,
+  });
+
+  // Output the keys in Base64 format for .env storage
+  console.log(`XMTP_KEY_BUNDLE=${Buffer.from(keys).toString(ENCODING)}`);
+
+  // Example of how you might use these keys in creating the client
+  const client = await Client.create(null, {
+    ...clientOptions,
+    privateKeyOverride: keys,
+  });
+
+  return client;
+};
+
+// Example usage
+const main = async () => {
+  const privateKey = "<YOUR_PRIVATE_KEY>"; // Replace with your private key
+  const wallet = new Wallet(privateKey);
+
+  console.log(`Using Wallet Address: ${wallet.address}`);
+
+  await generateAndPrintKeys(wallet);
+};
+
+main().catch(console.error);
+```
+
+### Run the script
+
+Run the script to get the key bundle. Make a note of the key bundle. You’ll need it in the next step.
+
+```bash
+tsx generateAndPrintKeys.ts
+```
+
+## Set up the environment
+
+In the `broadcast-example-api` directory, create an `.env` file with the following contents, for example.
+
+Replace `<your hex string of broadcast key bundle>` with the key bundle you got from the `generateAndPrintKeys.ts` script.
+
 ```
 XMTP_ENV=dev
 PORT=6989
 
-# XMTP Broadcast specific configuration
+# XMTP Broadcast-specific configuration
 XMTP_RATE_LIMIT_AMOUNT=1000
 XMTP_RATE_LIMIT_DURATION=300000
-XMTP_KEY_BUNDLE=<hex string of broadcast keybundle>
+XMTP_KEY_BUNDLE=<your hex string of broadcast key bundle>
 XMTP_FILE_PERSISTENCE_PATH=/tmp/xmtp
 ```
+
+### Update the CORS policy
+
+In `index.ts` at the root of the `broadcast-example-api` directory, add `"http://localhost:3000"` as shown in the following example.
+
+This addition enables the broadcast API to accept requests from `http://localhost:3000` , which will be the URL of the example subscribe-broadcast page that you’ll use with the broadcast API in an upcoming step.
+
+```tsx
+app.use(
+  cors({
+    origin: ["http://localhost:3000", "https://subscribe-broadcast.vercel.app"],
+  })
+);
+```
+
+## Install dotenv
+
+If not already installed, install dotenv.
+
+```bash
+yarn add dotenv
+```
+
+### Load environment variables to your app
+
+At the top of the `src/index.ts` file, add the following line. This will load all variables from your `.env` file into `process.env`.
+
+```tsx
+import "dotenv/config";
+```
+
+## Install and start Redis
+
+If not already installed, install Redis.
+
+### Install Redis
+
+```bash
+brew install redis
+```
+
+### Start Redis
+
+```bash
+redis-server
+```
+
+### Verify Redis is running
+
+```bash
+redis-cli ping
+```
+
+## Run the broadcast API
+
+```bash
+yarn dev
+```
+
+To use this example broadcast API to try out the flow to subscribe to a broadcast with built-in consent, use it with the [subscribe-broadcast repo](https://github.com/xmtp/subscribe-broadcast).
+
+To learn how, see [Try the subscribe with built-in consent and broadcast flow](https://docs.xmtp.org/consent/subscribe-broadcast).
```

Comment:
How do https://www.npmjs.com/package/@xmtp/broadcast-sdk and https://github.com/alexrisch/broadcaster-app come into the picture? Is there an opportunity to surface them in this README?